Runtime/Components/CinemachineSplineDolly.cs

```diff
@@ -0,0 +1,399 @@
+        void OnValidate()
+        {
+            if (m_TrackCache != null)
+            {
+                m_TrackCache.Spline.changed -= onSplineChanged;
+                m_TrackCache = m_Track;
+                m_Registered = false;
+            }
+            if (!m_Registered && m_Track != null && m_Track.Spline != null)
+            {
+                m_Registered = true;
+                m_TrackCache = m_Track;
+                m_Track.Spline.changed += onSplineChanged;
+            }
+
+            m_Damping.x = Mathf.Clamp(m_Damping.x, 0, 20);
+            m_Damping.y = Mathf.Clamp(m_Damping.y, 0, 20);
+            m_Damping.z = Mathf.Clamp(m_Damping.z, 0, 20);
+            m_RotationDamping.x = Mathf.Clamp(m_RotationDamping.x, 0, 20);
+            m_RotationDamping.y = Mathf.Clamp(m_RotationDamping.y, 0, 20);
+            m_RotationDamping.z = Mathf.Clamp(m_RotationDamping.z, 0, 20);
+        }
+
+        void UpdateSplineRollOverrideCache()
+        {
+            m_RollOverrideExtension = null;
+            // if vcam has an override, use that
+            if (transform.parent.TryGetComponent(out m_RollOverrideExtension))
+            {
+#if UNITY_EDITOR
+                m_RollOverrideExtension.splineContainer = m_Track; // this is needed by the RollHandle to work in the scene view
+#endif
+            }
+            // else if the spline has an override, use that
+            else if (m_Track.TryGetComponent(out m_RollOverrideExtension)) {}
+        }
+
+        CinemachineSplineRollOverrideExtension m_RollOverrideExtension; // don't use this directly
+        CinemachineSplineRollOverrideExtension RollOverrideExtension
+        {
+            get
+            {
+                if(m_RollOverrideExtension == null)
+                    UpdateSplineRollOverrideCache();
+
+                return m_RollOverrideExtension;
+            }
+        }
+        /// <summary>Positions the virtual camera according to the transposer rules.</summary>
+        /// <param name="curState">The current camera state</param>
+        /// <param name="deltaTime">Used for damping.  If less that 0, no damping is done.</param>
+        public override void MutateCameraState(ref CameraState curState, float deltaTime)
+        {
+            if (!IsValid)
+            {
+                m_PathPosition = 0;
+                return;
+            }
+
+            // splines work with normalized position by default, so we convert m_PathPosition to normalized at the start
+            var pathSpline = m_Track.Spline;
+            if (pathSpline == null || pathSpline.Count == 0)
+            {
+                return;
+            }
+            if (pathSpline.Count == 1)
+            {
+                curState.RawPosition = m_PreviousCameraPosition = pathSpline[0].Position;
+                curState.RawOrientation = m_PreviousOrientation = pathSpline[0].Rotation;
+                return;
+            }
+            
+            m_PathPosition = 
```

Comment:
style comment: the code is correct but it would feel better to me if you did not modify m_PathPosition here.  Why not instead just declare var newPathPosition = m_PathPosition here, and then always modify newPathPosition?